### ISSUE TITLE
Update of validation process step 4 (final step)

### ISSUE THREAD

**Issue by AnneV-Learn** [FreeUKGen/FreeCENMigration#1790]
When the freecen1_vld_entry_propagations collection was first created in was not envisaged that these pob propagations would be used by CSVProc data, so as the final step (Step 4) of the Strengthening of the Validation Process #1783 epic it would be advisable to rename the collection to freecen_pob_propagations to avoid future confusion.

This is likely to entail:

Renaming/migrating the freecen1_vld_entry_propagations collection data to freecen_pob_propagations and creating an index as was applied to freecen1_vld_entry_propagations.

Rename references in the associated model, views, controller and library entires (including Rakes) throughout all the project code.

**Comment by AnneV-Learn** [comment]
When the freecen1_vld_entry_propagations collection was first created in was not envisaged that these pob propagations would be used by CSVProc data, so as the final step (Step 4) of the Update of Validation Process epic it would be advisable to rename the collection to freecen_pob_propagations to avoid future confusion.

This is likely to entail:

Renaming/migrating the freecen1_vld_entry_propagations collection data to freecen_pob_propagations and creating an index as was applied to freecen1_vld_entry_propagations

Rename references in the associated model, views, controller and library entires (including Rakes) throughout all the project code.

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
I am thinking it might be an idea to also add  a 'Manage POB Propagation Records' Action in 'Your Actions' where POB Propagations could be viewed by anyone(?) and deleted by Data Manager or System Administrator (for hopefully the rare case were propagations are created by mistake)

**Comment by geoffj-FUG** [comment]
Anne

Yes please. I have been working blind with what is in the propagation data. That will be  very useful.

It should be read only for Validators and Coordinators and read / write for the Data Manager and System Administrator.

Can we make it sortable by county?

Geoff

**Comment by AnneV-Learn** [comment]
@geoff I recommend that we delay this until the new CSV POB validation functionality (#1787 etc) has been live for at least a month, so that any 'teething' issues can be ironed out first.

**Comment by geoffj-FUG** [comment]
I am happy with that delay.
I have issued a preview of the changed instructions for validation and asked for feedback. One month should allow us to evaluate these instructions and any tweaking to do.
Once we are stable and the validation is up and going then we can implement this change.
Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
The renaming of the POB Propagations collection has been successfully implemented on Test3.
I have also added a '**Manage POB Propagations**' Action for County Coords, System Administrators and Data managers. It appears as an **Action** after having chosen the County to Manage. Only the POB Propagation records for Verbatim Birth County matching the County are displayed and the 'Show'  screen offers a Delete button. 
Not sure exactly what you want for 'Checkers/Validators' - just a View List of Propagations option as an action?.
Anyway, perhaps you can take a look and report back. Thanks.

**Comment by Vino-S** [comment]
@geoffj-FUG to test

**Comment by geoffj-FUG** [comment]
@AnneV-Learn 
This looks good. Delete action should be available to Coordinators, Data Manager and System Admin. Validators should have read only access.
If we have these permissions then we can deploy this story.
Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG  At the moment it is only available to roles that have access to the 'Manage Counties' as the display is restricted to Propagation data related to the specified county (when you choose which County to Manage), and so that does not include validators. 
Validators and Transcribers currently have access to the following: ['CSV Batches', 'Gazetteer', 'FreeCEN Handbook', 'Communicate', 'Profile', 'Roadmap', 'Display County Coordinators', 'Display Syndicate Coordinators', 'FreeUKGenealogy  Policies']

Maybe we need a different display for Validators (and Transcribers?)  where they pick the County and there is no Delete option?

**Comment by geoffj-FUG** [comment]
Anne

 

I am happy to deploy without a read only for validators. Transcribers do not require access.

 

We can give validators read only access if the coordinators request it.

 

Geoff

**Comment by AnneV-Learn** [comment]
Ok @geoffj-FUG. 
The deployment of this story is not straightforward as it involves renaming a Mongodb Collection. I'll contact Vino on Monday and work out with her when the deployment should be done.

**Comment by geoffj-FUG** [comment]
Thanks Anne

 

Geoff

**Comment by AnneV-Learn** [comment]
@Vino-S  - I have contacted you via Slack with the deployment steps for this

**Comment by AnneV-Learn** [comment]
This has now been deployed.

**Comment by DeniseColbert** [comment]
Anne to see if all aspects complete, then can close.

**Comment by AnneV-Learn** [comment]
@geoffj-FUG the modifications to the Propagations List and Details that you mentioned in the scrum are actually requested and coded against #1822, which says you have tested on Test 3 and is ready for deployment (as noted in the Scrum) so I'll close this story.

**Comment by geoffj-FUG** [comment]
Anne

 

The last requested tweak to the results is not live.

 

On the front page of the results can you make columns 3 and 4 the place name that the entry is changed to please?

 

The name and date of the original entry just need to be on the second page.

 

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG Yes as noted above you made the last request for a tweak to the Propagation List and Details against Issue #1822 so that is where it has been coded. When that branch goes live (Vino knows) those changes will go live too.

**Comment by geoffj-FUG** [comment]
I Typed in the wrong story! Sorry.

 

It should g live soon so OK.

 

Geoff
